qiskit/synthesis/clifford/clifford_decompose_greedy.py

```diff
@@ -32,22 +32,22 @@
 
 
 def synth_clifford_greedy(clifford):
-    """Decompose a Clifford operator into a QuantumCircuit based on the
-    greedy Clifford compiler that is described in Appendix A of
-    Bravyi, Hu, Maslov and Shaydulin.
+    """Decompose a :class:`.Clifford` operator into a :class:`.QuantumCircuit` based
+    on the greedy Clifford compiler that is described in Appendix A of
+    Bravyi, Hu, Maslov and Shaydulin [1].
```

Comment:
Dmitri Maslov said that perhaps we should better emphasize here that we only implemented the greedy Clifford compiler from Appendix A, and not the templates and symbolic Pauli gates optimizations that are mentioned in the same paper

Reply:
@ShellyGarion do you have any suggestion on how to clarify it more? I am not very familiar with the paper.

Reply:
In a nutshell: the paper describes several algorithms, while Qiskit implements only one of these. On the one hand, I believe that it's generally understood that a specific implementation of an algorithm does not necessarily implement all of the details mentioned in a paper, especially when the implementers and the authors are not the same people. On the other hand, given that both come from IBM, others do tend to make this confusion, and there were explicit papers that said something like this: "... our implementation is better than one described in paper [blah] because we have compared it against Qiskit's algorithm [blah]...". To avoid the confusion in this particular case, @ShellyGarion is suggesting to add an explicit comment "Remark: this function implements only the greedy Clifford compiler from Appendix A of [1], and not the templates and symbolic Pauli gates optimizations that are mentioned in the same paper."

Reply:
I've added a warning, maybe it's too flashy, I can also replace it with a "note":

<img width="722" alt="Screenshot 2024-02-12 at 16 03 00" src="https://github.com/Qiskit/qiskit/assets/57907331/cf417b35-d716-4889-be61-f620f77b0b69">

Reply:
I think that a note is enough, thanks!